### PR TITLE
Clarify the linkerscript example

### LINKED ISSUE #1006: rgblink linker script format manual doesn't clearly explain that sections should be specified in quotes
https://rgbds.gbdev.io/docs/v0.5.2/rgblink.5

The provided words in quotes seem more like comments/explanations. It should be made clear that section names have to be provided in quotes, one per line. Ideally, an example of a well-formed linker script should be provided.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1006

### CODE REVIEW

File: man/rgblink.5 [hunk 1]
```diff
@@ -14,18 +14,21 @@
 .Sh DESCRIPTION
 The linker script is an external file that allows the user to specify the order of sections at link time and in a centralized manner.
 .Pp
-A linker script consists on a series of banks followed by a list of sections and, optionally, commands.
-They can be lowercase or uppercase, it is ignored.
+A linker script consists of a series of bank declarations, each optionally followed by a list of section names (in double quotes) or commands.
+All reserved keywords (bank types and command names) are case-insensitive;
+all section names are case-sensitive.
```
Comment (ISSOtm) [inline man/rgblink.5:19]:
```suggestion
All reserved keywords (bank types and command names) are case-insensitive; all section names are case-sensitive.
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
Should we also clarify the accepted charset in strings? (In particular, last I checked, no escapes are supported.)

**Comment by Rangi42** [comment]
> Should we also clarify the accepted charset in strings? (In particular, last I checked, no escapes are supported.)

I see backslash-escape code in src/link/script.c for `\n`, `\r`, `\t`, `\"`, `\\` in `"strings"`.

**Comment by ISSOtm** [comment]
I might've added those escape sequences for SDCC interop. I think. Whatever.

I think it should be specified that the same escape sequences are supported that RGBASM does in strings, for the obvious parity reason. (If both fall in sync, then, it'll be a bug.)

**Review by ISSOtm — APPROVED** [review]
Awesome, thank you very much!

**Pull request merged**